### PR TITLE
DevMojo invokes other goals only if those are actually configured

### PR DESCRIPTION
**Pull request by famod**
Includes a fix to DevMojoIT.testCommandMode* when ANSI escape sequences are logged.

Came up here: https://github.com/quarkusio/quarkus/issues/13671#issuecomment-738336346

### CODE REVIEW

File: integration-tests/maven/src/test/java/io/quarkus/maven/it/DevMojoIT.java [hunk 1]
```diff
@@ -905,4 +904,16 @@ public void testThatDependencyInParentIsEvaluated() throws IOException, MavenInv
...
     }
+
+    @Test
+    public void testThatGenerateCodeGoalIsNotTriggeredIfNotConfigured() throws IOException, MavenInvocationException {
+        testDir = initProject("projects/classic-no-generate");
+        // the skip parameter triggers a log statement by the generate goal,
+        // otherwise there would be no way to tell from the logfile that the goal was invoked
+        runAndCheck(false, "-Dquarkus.generate-code.skip=true");
+
+        assertThat(running.log()).doesNotContain("Skipping Quarkus code generation");
```
Comment (famod) [inline integration-tests/maven/src/test/java/io/quarkus/maven/it/DevMojoIT.java:915]: Not ideal, but I did not find a better way to check whether the `generate-code` is actually invoked.

**Review by aloubyansky — APPROVED** [review]
Thanks @famod

**Comment by famod** [comment]
Formatting fixed (hopefully).

**Comment by aloubyansky** [comment]
The failure isn't related.

**Pull request merged**